external/storm-redis/src/main/java/org/apache/storm/redis/state/RedisKeyValueState.java

```diff
@@ -187,70 +206,73 @@ public void prepareCommit(long txid) {
                 LOG.debug("Nothing to save for prepareCommit, txid {}.", txid);
             }
             txIds.put(PREPARE_TXID_KEY, String.valueOf(txid));
+
             commands.hmset(txidNamespace, txIds);
-            pendingCommit = Collections.unmodifiableMap(currentPending);
+            pendingCommit = Maps.unmodifiableNavigableMap(currentPending);
         } finally {
-            jedisContainer.returnInstance(commands);
+            container.returnInstance(commands);
         }
     }
 
     @Override
     public void commit(long txid) {
         LOG.debug("commit txid {}", txid);
         validateCommitTxid(txid);
-        JedisCommands commands = null;
+        RedisCommands commands = null;
         try {
-            commands = jedisContainer.getInstance();
+            commands = container.getInstance();
             if (!pendingCommit.isEmpty()) {
-                List<String> keysToDelete = new ArrayList<>();
-                Map<String, String> keysToAdd = new HashMap<>();
-                for(Map.Entry<String, String> entry: pendingCommit.entrySet()) {
-                    if (RedisEncoder.TOMBSTONE.equals(entry.getValue())) {
-                        keysToDelete.add(entry.getKey());
+                List<byte[]> keysToDelete = new ArrayList<>();
+                Map<byte[], byte[]> keysToAdd = new HashMap<>();
+                for(Map.Entry<byte[], byte[]> entry: pendingCommit.entrySet()) {
+                    byte[] key = entry.getKey();
+                    byte[] value = entry.getValue();
+                    if (Arrays.equals(DefaultStateEncoder.TOMBSTONE, value)) {
+                        keysToDelete.add(key);
                     } else {
-                        keysToAdd.put(entry.getKey(), entry.getValue());
+                        keysToAdd.put(key, value);
                     }
                 }
                 if (!keysToAdd.isEmpty()) {
                     commands.hmset(namespace, keysToAdd);
                 }
                 if (!keysToDelete.isEmpty()) {
-                    commands.hdel(namespace, keysToDelete.toArray(new String[0]));
+                    commands.hdel(namespace, keysToDelete.toArray(new byte[0][]));
                 }
             } else {
                 LOG.debug("Nothing to save for commit, txid {}.", txid);
             }
             txIds.put(COMMIT_TXID_KEY, String.valueOf(txid));
             commands.hmset(txidNamespace, txIds);
             commands.del(prepareNamespace);
-            pendingCommit = Collections.emptyMap();
+            pendingCommit = createPendingCommitMap();
```

Comment:
nit: can be `Collections.emptyNavigableMap()` (java8) or a static final variable initialized with an empty TreeMap